### ISSUE TITLE
VSCode debugger issues

### ISSUE THREAD

**Issue by czrptr** [mike-lischke/vscode-antlr4#147]
I can't get the action file to work.

When I launch the debugger I get this error:
```
Could not launch debug session:

ReferenceError: module is not  defined
```

**.vscode/launch.json**
```json
// ...
    {
        "name": "Debug ANTLR4 grammar",
        "type": "antlr-debug",
        "request": "launch",
        "input": "${workspaceFolder}/Grammar/test.txt",
        "grammar": "${workspaceFolder}/Grammar/MirParser.g4",
        "actionFile": "${workspaceFolder}/Grammar/actions.js",
        "startRule": "start",
        "printParseTree": true,
        "visualParseTree": true,
    }
// ...
```

**Grammar/MirParser.g4**
```antlr
// ...
start : test ;

test
  : {notSawMinus}? OpAdd test
  | OpSub test { notSawMinus = false; }
  |
  ;
// ...
```

**Grammar/actions.js**
```js
"use strict";

var notSawMinus = true;

module.exports.evaluateLexerPredicate =
function (parser, ruleIndex, predIndex, predicate) {
    return eval(predicate);
};

module.exports.evaluateParserPredicate =
function (parser, ruleIndex, predIndex, predicate) {
    return eval(predicate);
};
```

The debugger works fine without the `"actionFile"` option in **launch.json** and without the predicates inside **MirParser.g4**. What I want the clarify is that just this feature seems to not be working.

Is there a solution to this problem?

**Comment by mike-lischke** [comment]
Looks ok to me, except that `{ notSawMinus = false; }` won't be executed, because the debugger does not execute any action code. The error message is not very descriptive, so it's tricky to say what's wrong. Have you try the demo code from https://github.com/mike-lischke/vscode-antlr4/blob/master/doc/grammar-debugging.md?

**Comment by chris-miner** [comment]
I've tried working through the example on the grammar-debugging page, but have encountered the same problem.  The error message is:
```
Could not prepare debug session:

ReferenceError: module is not defined
```
The error has to do with the `module.exports.evaluate...` lines.

**Comment by chris-miner** [comment]
I can't actually tell if the actions work without the module.exports bit.  I downloaded the HEAD version of the plug-in and based on logging I added to the sample doesItBlend() function found that the predicate is evaluated without the extra module.exports bit.  I wanted to try that on the 2.2.4 tag, but that didn't build for me.

**Comment by chris-miner** [comment]
So now that I have a copy of 2.2.4 running, I see that this problem is addressed in two other issues take your pick [#129](https://github.com/mike-lischke/vscode-antlr4/issues/117#issuecomment-711800100) or [#129](https://github.com/mike-lischke/vscode-antlr4/issues/129#issuecomment-805587481)

**Comment by mike-lischke** [comment]
I just released version 2.2.5, with simplified action code handling. See if that works better now for you.
